### PR TITLE
Support node in wasm package

### PR DESCRIPTION
**Pull request by mayank99**
As discussed in #386, will try to read from filesystem (using `fs`) if used in a node environment, and fallback to `fetch` for browser/deno environments.

### CODE REVIEW

**Comment by shyim** [comment]
It would be nice if we could just pass the file to that function. So it's abstract and anyone can fetch the file from they want :)

**Comment by LeoDog896** [comment]
Also considering the fact that it doesn't work on node or the browser anyways (#394).

**Comment by mayank99** [comment]
okay, promoting this out of draft since there is demand for it

@devongovett do let me know if you would like to see a different api (based on the [suggestion](https://github.com/parcel-bundler/lightningcss/pull/387#issuecomment-1378352048) above maybe?).

File: wasm/index.mjs [hunk 1]
```diff
@@ -51,3 +51,12 @@ async function load(module, imports) {
     }
   }
 }
+
+async function fetchOrReadFromFs(inputPath) {
+  try {
+    const fs = await import('fs');
+    return fs.readFileSync(inputPath);
```
Comment (devongovett) [inline wasm/index.mjs:58]: Would we need to use a different API for Deno? Or can Deno `fetch` a `file://` url?

Comment (LeoDog896) [reply wasm/index.mjs:58]: I do know that deno can instantiate streaming ([full docs](https://deno.land/manual@v1.12.2/getting_started/webassembly)), but I'm not sure about the consequences of deno's design are.

Comment (mayank99) [reply wasm/index.mjs:58]: well, i spun up a deno deploy playground and tried to `await import('fs')` and got this error (as expected)

```
TypeError: Relative import path "fs" not prefixed with / or ./ or ../ from "file:///src/main.ts"
```

after that, i wondered if it currently even works in deno so i tried to run the snippet from the [docs](https://lightningcss.dev/docs.html#from-deno-or-in-browser) and got a similar error:

```
Relative import path "napi-wasm" not prefixed with / or ./ or ../ from
```

i then got it to work after i added an [import map](https://deno.land/manual@v1.29.4/basics/modules/import_maps):
```json
{
  "imports": {
    "napi-wasm": "https://unpkg.com/napi-wasm"
  }
}
```

then i tested passing a full url to `init` and got it work

```js
import { serve } from 'https://deno.land/std@0.155.0/http/server.ts';
import init, { transform } from 'https://unpkg.com/lightningcss-wasm';

await init('https://unpkg.com/lightningcss-wasm/lightningcss_node.wasm');

let {code, map} = transform({
  filename: 'style.css',
  code: new TextEncoder().encode('.foo { color: red }'),
  minify: true,
});

const css = new TextDecoder().decode(code);

serve((req: Request) => new Response(css)); // .foo{color:red}
```

all of that to say, my current PR should be fine as long as the import map is added.

**Review by devongovett — APPROVED** [review]

**Pull request merged**